### PR TITLE
Microsoft-graph-toolkit v1.1 documentation update

### PR DESCRIPTION
**Pull request by vogtn**
This PR facilitates a documentation update for the 
release of microsoft-graph-toolkit version 1.1

Covers Components and Features:

- **mgt-get** creation of documentation, this component allows the developer to define custom graph queries
- **proxy-provider** creation of documentation, allows developer to define custom proxy for routing provider. 
- **mgt-agenda** updated docs, eventClick, if user clicks on an "event" will fire. 
- **people-picker** updated docs, load and error templates 
- **person-card** updated docs, inherit details from parent components
- **people** updated docs, new property "group-id" to define specific graph group.
- **tasks** updated docs, new templates and taskClick event
- **templates** updated docs, $parent and $index features added. 

Reference:
- [x] mgt-get  0848b37
- [x] eventClick event on Agenda b6d40d2
- [x] loading and the error templates in people picker f7a8343
- [x] inherit-details attribute in person-card 93e41ee
- [x] group-id attribute in people aa732ca
- [x] browser support b7400bf
- [x] mgt-tasks 293215c
- [x] ProxyProvider c4ea853 
- [x] template support for $index and $parent 92f67a5
- [x] person-card template context update 93e41ee (inherit details)
- [ ] link to sharepoint sample readme from sharepoint provider docs 
- [ ] people-picker template for selected-people 

Clears the following issue: 
https://github.com/microsoftgraph/microsoft-graph-toolkit/issues/184

### CODE REVIEW

**Comment by nmetulev** [comment]
@vogtn, make sure to address the suggestions above

File: concepts/toolkit/components/get.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: "Get component in the Microsoft Graph Toolkit"
+description: "A Get component allows you to make any GET query from Microsoft Graph directly in your HTML."
+localization_priority: Normal
+author: nmetulev
+---
+
+# Login component in the Microsoft Graph Toolkit
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]: Title is wrong - should be get component

File: concepts/toolkit/components/get.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    loading
+  </template>
+</mgt-get>
+```
+
+#### Properties and Attributes
+
+The only required attribute is `resource`. However, you can use several properties to change the behavior.
+
+| Property | Attribute | Description |
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]: Can you swap property and attribute so attributes are first (also for all the other docs)?

File: concepts/toolkit/components/get.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+| --- | --- | --- |
+| `resource` | `resource` | The resource to get from Microsoft Graph (ex: `/me`) |
+| `scopes` | `scopes` | Optional array of strings if using the property or a comma delimited scopes if using the attribute. The component will use these scopes (with a supported provider) to ensure the user has  consented to the right permission |
+| `version` | `version` | Optional API version to use when making the GET request. Default is `v1.0`.  |
+| `maxPages` | `max-pages` | Optional number of pages (for resources that support paging). Default is 3. Setting this value to 0 will GET all pages  |
+| `response` | N/A | The raw response from Microsoft Graph  |
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]:
```suggestion
| `response` | N/A | Read only response from Microsoft Graph if the request was successful  |
```

File: concepts/toolkit/components/get.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+| `error` | N/A | The raw error from Microsoft Graph  |
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]:
```suggestion
| `error` | N/A | Read only error from Microsoft Graph if the request was not successful |
```

File: concepts/toolkit/components/get.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+| --- | --- | --- |
+| `dataChange` | The detail contains the `response` and `error` objects | Fired when the response or error change |
+
+#### Templates
+
+The `mgt-get` component supports several [templates](../templates.md) that you can use to define the look and feel. To specify a template, include a `<template>` element inside a component and set the `data-type` value to one of the following.
+
+| Data type | Data context | Description |
+| --- | --- | --- |
+| `default` | a raw response from Microsoft Graph | The default template is required to render the data coming from Microsoft Graph  |
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]:
```suggestion
| `default` | the response from Microsoft Graph | The default template is required to render the data coming from Microsoft Graph  |
```

File: concepts/toolkit/components/get.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+| `error` | the raw error from Microsoft Graph | This template will be used if there is an error making the request |
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]:
```suggestion
| `error` | the error from Microsoft Graph | This template will be used if there is an error making the request |
```

File: concepts/toolkit/components/get.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+| `loading` | N/A | This template is used while the request is being made |
+
+## Microsoft Graph permissions
+
+This control will require Microsoft Graph APIs and permissions based on the `resource` given.
+
+Example:
```
Comment (nmetulev) [inline concepts/toolkit/components/get.md]: Remove example, and simply link to the permission reference in the docs: https://docs.microsoft.com/graph/permissions-reference

File: concepts/toolkit/components/people-picker.md [hunk 8]
```diff
@@ -81,14 +81,34 @@ mgt-people-picker {
...
 
+## Templates
+
+ `mgt-people-picker` supports several [templates](../templates.md) that you can use to replace certain parts of the component. To specify a template, include a `<template>` element inside a component and set the `data-type` value to one of the following.
+
+| Data type | Data context | Description |
+| --- | --- | --- |
+| `loading` | `null`: no data | The template used to render the state of picker while request to graph is being made. |
+| `error` | `null`: no data| The template used if user search returns no users |
+| `selected-user` | `person`: Microsoft graph user| The template used if user search returns no users |
```
Comment (nmetulev) [inline concepts/toolkit/components/people-picker.md]: Description is a copy and paste from previous

File: concepts/toolkit/components/tasks.md [hunk 9]
```diff
@@ -89,6 +91,35 @@ mgt-tasks {
 }
 ````
 
+#### Events
+| Event | Detail | Description |
+| --- | --- | --- |
+| `taskClick` | The detail contains the respective `task` object | Fired when the user interacts via `click` on a task. |
```
Comment (nmetulev) [inline concepts/toolkit/components/tasks.md]:
```suggestion
| `taskClick` | The detail contains the respective `task` object | Fired when the user clicks or taps on a task |
```

File: concepts/toolkit/components/tasks.md [hunk 10]
```diff
@@ -89,6 +91,35 @@ mgt-tasks {
...
+
+## Templates
+
+The `tasks` component supports several [templates](../templates.md) that allow you to replace certain parts of the component. To specify a template, include a `<template>` element inside a component and set the `data-type` value to one of the following:
+
+| Data type     | Data context              | Description                                                       |
+| ---------     | ------------------------- | ----------------------------------------------------------------- |
+| task     | task: contains all details pertaining to task | replaces the whole default task. |
```
Comment (nmetulev) [inline concepts/toolkit/components/tasks.md]:
```suggestion
| task     | task: a planner or to-do task object | replaces the whole default task. |
```

File: concepts/toolkit/components/tasks.md [hunk 11]
```diff
@@ -89,6 +91,35 @@ mgt-tasks {
...
+| task-details | task: contains all details pertaining to task | template replaces the details section of the task. |
```
Comment (nmetulev) [inline concepts/toolkit/components/tasks.md]:
```suggestion
| task-details | task: a planner or to-do task object | template replaces the details section of the task. |
```

File: concepts/toolkit/overview.md [hunk 12]
```diff
@@ -196,3 +196,9 @@ Finally, use the component as you normally would in your template.
 ```html
 <mgt-person [personDetails]="person" show-name></mgt-person>
 ```
+## Browser Support
+
+
+| ![IE/ Edge](images/edgeIcon.png)<br>IE / Edge | ![Firefox](images/firefoxIcon.png)<br>Firefox | ![Chrome](images/chromeIcon.png)<br>Chrome | ![Opera](images/operaIcon.png)<br>Opera |
+| --------- | --------- | --------- | --------- | --------- | --------- | --------- |
+| Edge, Dev|  Latest Version| Latest Version| Lastest Version|
```
Comment (nmetulev) [inline concepts/toolkit/overview.md]: Should we also add internet explorer and the mobile browsers?

File: concepts/toolkit/templates.md [hunk 13]
```diff
@@ -63,6 +63,30 @@ This format can also be used inside of attributes:
...
 > **Note:** You can also expand objects such as `{{event}}` and they will render as JSON strings. This can be useful when you're developing the templates.
 
+**Context**
+
+There are also accessible properties when binding, for template context. 
+
+| Property |  Description |
+| --- | --- | --- |
+| `$index` | numerical index of item being rendered while being looped with `data-for` |
+| `$parent` | If a template is rendered inside of another template, the user is able to find data context of the parent within the child template. |
```
Comment (nmetulev) [inline concepts/toolkit/templates.md]:
```suggestion
| `$parent` | If a template is rendered inside of another template, the `$parent` property allows the developer to access the parent data context. |
```

File: concepts/toolkit/templates.md [hunk 14]
```diff
@@ -63,6 +63,30 @@ This format can also be used inside of attributes:
 
 > **Note:** You can also expand objects such as `{{event}}` and they will render as JSON strings. This can be useful when you're developing the templates.
 
+**Context**
```
Comment (nmetulev) [inline concepts/toolkit/templates.md]:
```suggestion
## Data Context helper properties
```

File: concepts/toolkit/templates.md [hunk 15]
```diff
@@ -63,6 +63,30 @@ This format can also be used inside of attributes:
...
+
+There are also accessible properties when binding, for template context.
```
Comment (nmetulev) [inline concepts/toolkit/templates.md]:
```suggestion
The following properties can also be used with the data context object in your templates
```

File: concepts/toolkit/templates.md [hunk 16]
```diff
@@ -63,6 +63,30 @@ This format can also be used inside of attributes:
...
+
+There are also accessible properties when binding, for template context. 
+
+| Property |  Description |
+| --- | --- | --- |
+| `$index` | numerical index of item being rendered while being looped with `data-for` |
+| `$parent` | If a template is rendered inside of another template, the user is able to find data context of the parent within the child template. |
+
+```html
+<mgt-person>
```
Comment (nmetulev) [inline concepts/toolkit/templates.md]: This example doesn't really make sense. You can use this example:

```html
<span data-for="language in languages">
    {{ language.displayName }}<span data-if="$index < languages.length - 1">, </span>
</span>
```

Comment (vogtn) [reply concepts/toolkit/templates.md]: [Updated example](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/826c791e0854be1b0dfc11dcc6147413c40b9d8d#diff-880fdc26a24a1a1eebafb18a328632a6R79-R81)

File: concepts/toolkit/templates.md (hunk 15 above)
Comment (vogtn) [reply concepts/toolkit/templates.md]: [updated](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/826c791e0854be1b0dfc11dcc6147413c40b9d8d#diff-880fdc26a24a1a1eebafb18a328632a6R67-R68)

File: concepts/toolkit/overview.md (hunk 12 above)
Comment (vogtn) [reply concepts/toolkit/overview.md]: Added separate section for IE11, and added mobile support for chrome in the versioning section. 
Not sure if we should officially declare support for ios safari and opera mini as they are still developing overall support for Custom elements and things could change soon. The Samsung browser seems to work well however. 
c9cdc16

File: concepts/toolkit/templates.md (hunk 13 above)
Comment (vogtn) [reply concepts/toolkit/templates.md]: [updated context description](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/826c791e0854be1b0dfc11dcc6147413c40b9d8d#diff-880fdc26a24a1a1eebafb18a328632a6R73)

File: concepts/toolkit/components/get.md (hunk 1 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: Whoops sorry... Here's the updated title:
https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-8912736507b3669239468443fb54bf44R8

File: concepts/toolkit/components/get.md (hunk 2 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: Updated following components for swapping properties and attributes:

+GET
+Agenda
+Login
+People-picker
+people
+Person
+tasks

File: concepts/toolkit/components/get.md (hunk 6 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: (updated)[https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-8912736507b3669239468443fb54bf44R50]

File: concepts/toolkit/templates.md (hunk 14 above)
Comment (vogtn) [reply concepts/toolkit/templates.md]: [updated title](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-880fdc26a24a1a1eebafb18a328632a6R66)

File: concepts/toolkit/components/tasks.md (hunk 11 above)
Comment (vogtn) [reply concepts/toolkit/components/tasks.md]: updated (task-details)[https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-31321da30579c0ee0299430f0959f893R107]

File: concepts/toolkit/components/tasks.md (hunk 10 above)
Comment (vogtn) [reply concepts/toolkit/components/tasks.md]: updated [task](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-31321da30579c0ee0299430f0959f893R106)

File: concepts/toolkit/components/tasks.md (hunk 9 above)
Comment (vogtn) [reply concepts/toolkit/components/tasks.md]: updated [taskClick](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/826c791e0854be1b0dfc11dcc6147413c40b9d8d#diff-31321da30579c0ee0299430f0959f893R97)

File: concepts/toolkit/components/people-picker.md (hunk 8 above)
Comment (vogtn) [reply concepts/toolkit/components/people-picker.md]: sorry... [updated](https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/files#diff-7e91f257372c955ad01ad9497529ed63R92)

File: concepts/toolkit/components/get.md (hunk 3 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: fixed in latest commit:
https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/9e2abbf77c2786ec7c6fb4ba6f13f114329f0a43

File: concepts/toolkit/components/get.md (hunk 4 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: fixed in latest commit: https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/9e2abbf77c2786ec7c6fb4ba6f13f114329f0a43

File: concepts/toolkit/components/get.md (hunk 5 above)
Comment (vogtn) [reply concepts/toolkit/components/get.md]: fixed in latest commit: fixed in latest commit: https://github.com/microsoftgraph/microsoft-graph-docs/pull/6425/commits/9e2abbf77c2786ec7c6fb4ba6f13f114329f0a43

File: concepts/toolkit/components/get.md (hunk 2 above)
Comment (nmetulev) [reply concepts/toolkit/components/get.md]: person still needs to be updated

**Pull request merged**